Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfUtility.cs

```diff
@@ -0,0 +1,370 @@
+                    //            gltfObject.materials[i].name == extensionRequired.Key)
+                    //        {
+                    //            gltfObject.materials[i].Extensions.Add(gltfObject.extensionsUsed[i], extensionRequired.Value);
+                    //            var extension = JsonUtility.FromJson<KHR_Materials_PbrSpecularGlossiness>(extensionRequired.Value);
+                    //            extension.ElementName = gltfObject.materials[i].name;
+                    //            gltfObject.RegisteredExtensions.Add(extension);
+                    //        }
+                    //    }
+                    //}
+                    //else
+                    {
+                        Debug.LogWarning($"Unsupported Extension: {gltfObject.extensionsRequired[i]}");
+                        return null;
+                    }
+                }
+            }
+
+            for (int i = 0; i < gltfObject.extensionsUsed?.Length; i++)
+            {
+                var extensionsUsed = GetGltfExtensionObjects(jsonString, gltfObject.extensionsUsed[i]);
+
+                foreach (var extensionUsed in extensionsUsed)
+                {
+                    // TODO Update this after KHR_materials_pbrSpecularGlossiness extension is supported
+                    //if (gltfObject.extensionsUsed[i].Equals("KHR_materials_pbrSpecularGlossiness"))
+                    //{
+                    //    for (int j = 0; j < gltfObject.materials.Length; j++)
+                    //    {
+                    //        if (!string.IsNullOrEmpty(gltfObject.materials[i].name) &&
+                    //            gltfObject.materials[i].name == extensionUsed.Key)
+                    //        {
+                    //            gltfObject.materials[i].Extensions.Add(gltfObject.extensionsUsed[i], extensionUsed.Value);
+                    //            var extension = JsonUtility.FromJson<KHR_Materials_PbrSpecularGlossiness>(extensionUsed.Value);
+                    //            extension.ElementName = gltfObject.materials[i].name;
+                    //            gltfObject.RegisteredExtensions.Add(extension);
+                    //        }
+                    //    }
+                    //}
+                    //else
+                    {
+                        Debug.LogWarning($"Unsupported Extension: {gltfObject.extensionsUsed[i]}");
+                    }
+                }
+            }
+
+            var meshPrimitiveAttributes = GetGltfMeshPrimitiveAttributes(jsonString);
+
+            for (int i = 0; i < gltfObject.meshes.Length; i++)
+            {
+                for (int j = 0; j < gltfObject.meshes[i].primitives.Length; j++)
+                {
+                    gltfObject.meshes[i].primitives[j].Attributes = JsonUtility.FromJson<GltfMeshPrimitiveAttributes>(meshPrimitiveAttributes[0]);
+                    meshPrimitiveAttributes.Remove(meshPrimitiveAttributes[0]);
```

Comment:
nit: I'd prefer if meshPrimitiveAttributes was cleared after looping compared to making these continual remove calls. It's not super intuitive whats going on by just always referencing the 0 index item

Reply:
picking up this fix, looks like meshPrimitiveAttributes is declared on the function stack so there's no reason to clear